x-pack/plugins/threat_intelligence/public/modules/timeline/hooks/use_investigate_in_timeline.ts

```diff
@@ -42,17 +42,23 @@ export const useInvestigateInTimeline = ({
   const securitySolutionContext = useContext(SecuritySolutionContext);
 
   const { key, value } = getIndicatorFieldAndValue(indicator, RawIndicatorFieldId.Name);
-  if (!fieldAndValueValid(key, value)) {
+  const sourceEventField = IndicatorFieldEventEnrichmentMap[key];
+
+  if (!fieldAndValueValid(key, value) || !sourceEventField) {
     return {} as unknown as UseInvestigateInTimelineValue;
   }
 
-  const dataProviders: DataProvider[] = [...IndicatorFieldEventEnrichmentMap[key], key].map(
```

Comment:
I believe this bug shows that we should be more strict with the types as we rely on a very specific set of fields for this feature